src/types/redis_stream.cc

```diff
@@ -343,8 +346,26 @@ rocksdb::Status Stream::DeletePelEntries(const Slice &stream_name, const std::st
     std::string entry_key = internalPelKeyFromGroupAndEntryId(ns_key, metadata, group_name, id);
     std::string value;
     s = storage_->Get(rocksdb::ReadOptions(), stream_cf_handle_, entry_key, &value);
+
     if (s.ok()) {
       *acknowledged += 1;
+      StreamPelEntry pel_entry = decodeStreamPelEntryValue(value);
+
+      const std::string &consumer_name = pel_entry.consumer_name;
+      std::string consumer_key = internalKeyFromConsumerName(ns_key, metadata, group_name, consumer_name);
+      std::string get_consumer_value;
+      s = storage_->Get(rocksdb::ReadOptions(), stream_cf_handle_, consumer_key, &get_consumer_value);
+      if (!s.ok() && !s.IsNotFound()) {
+        return s;
+      }
+      if (s.IsNotFound()) {
+        return rocksdb::Status::OK();
+      }
+
+      StreamConsumerMetadata consumer_metadata = decodeStreamConsumerMetadataValue(get_consumer_value);
+      consumer_metadata.pending_number -= 1;
```

Comment:
@Yangsx-1 , It seems that we should also update `pending_number` in `StreamConsumerMetadata`, when `xack` id, currently, we only update `pending_number` in `StreamConsumerGroupMetadata`.

Reply:
Good catch! You're right.

Reply:
Has this patch about `add support of XPENDING`?
If not can you move the change to a new PR? @LiuYuHui